### PR TITLE
ref(ts) Convert ProjectBadge to typescript

### PR DESCRIPTION
**Pull request by markstory**
I had to make a few small logic/typing changes but I think they are low risk changes.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/alerts/details/header.tsx [hunk 1]
```diff
@@ -124,12 +124,12 @@ export default class DetailsHeader extends React.Component<Props> {
...
                   <Projects slugs={[project]} orgId={params.orgId}>
-                    {({projects}) => (
-                      <ProjectBadge
-                        avatarSize={18}
-                        project={projects && projects.length && projects[0]}
-                      />
-                    )}
+                    {({projects}) =>
+                      projects &&
+                      projects.length && (
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/alerts/details/header.tsx:129]: Should we use optional chaining here?

**Review by matejminar — APPROVED** [review]
Looks good to me 🚀

File: src/sentry/static/sentry/app/components/idBadge/projectBadge.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import {AvatarProject} from 'app/types';
+import BaseBadge from 'app/components/idBadge/baseBadge';
+import BadgeDisplayName from 'app/components/idBadge/badgeDisplayName';
+
+type Props = {
+  project: AvatarProject;
+  avatarSize?: number;
+  className?: string;
+  hideOverflow?: boolean | string;
+  // Inherited from BaseBadge
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/idBadge/projectBadge.tsx:13]: Why can't you pull these props types in from BaseBadge?

Comment (markstory) [reply src/sentry/static/sentry/app/components/idBadge/projectBadge.tsx:13]: It hasn't been converted to typescript yet. It is next on my list though.

**Pull request merged**